runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/DataflowRunner.java

```diff
@@ -1192,12 +1186,8 @@ private ReadWithIds(UnboundedSource<T, ?> source) {
       @Override
       public final PCollection<ValueWithRecordId<T>> expand(PInput input) {
         return PCollection.<ValueWithRecordId<T>>createPrimitiveOutputInternal(
-            input.getPipeline(), WindowingStrategy.globalDefault(), IsBounded.UNBOUNDED);
-      }
-
-      @Override
-      protected Coder<ValueWithRecordId<T>> getDefaultOutputCoder() {
-        return ValueWithRecordId.ValueWithRecordIdCoder.of(source.getDefaultOutputCoder());
+            input.getPipeline(), WindowingStrategy.globalDefault(), IsBounded.UNBOUNDED)
+            .setCoder(ValueWithRecordId.ValueWithRecordIdCoder.of(source.getDefaultOutputCoder()));
```

Comment:
source.getDefaultOutputCoder() now can be null, right? I don't think you can pass a null coder to ValueWithRecordIdCoder.